wrapanapi/systems/redfish.py

```diff
@@ -83,11 +56,74 @@ def refresh(self):
     @property
     def name(self):
         """Return name from most recent raw data."""
-        return self._ems_ref
+        return self.raw.Id
+
+    @property
+    def description(self):
+        """Return description from most recent raw data."""
+        return self.raw.Description
 
     def uuid(self):
         """Return uuid from most recent raw data."""
-        return self._ems_ref
+        return self.raw.Id
+
+
+class RedfishServer(Server, RedfishResource):
+    state_map = {
+        'On': ServerState.ON,
+        'Off': ServerState.OFF,
+        'PoweringOn': ServerState.POWERING_ON,
+        'PoweringOff': ServerState.POWERING_OFF,
+    }
+
+    @property
+    def server_cores(self):
+        """Return the number of cores on this server."""
+        return sum(int(p.TotalCores) for p in self.raw.Processors.Members)
+
+    @property
+    def server_memory(self):
+        """Return the amount of memory on the server, in MiB."""
+        return self.raw.MemorySummary.TotalSystemMemoryGiB * 1024
+
+    @property
+    def state(self):
+        """Retrieve the current power status of the physical server."""
+        return self.raw.PowerState
+
+    def _get_state(self):
+        """
+        Return ServerState object representing the server's current state.
+
+        The caller should call self.refresh() first to get the latest status
+        from the API.
+        """
+        return self._api_state_to_serverstate(self.state)
+
+
+class RedfishChassis(PhysicalContainer, RedfishResource):
+    """API handler for this instance of the physical container."""
+
+    @property
+    def chassis_type(self):
+        """Retrieve the type of this chassis."""
+        return self.raw.ChassisType
+
+    @property
+    def led_state(self):
+        """Retrieve the status of the identifying LED on this chassis."""
+        if "IndicatorLED" in self.raw:
+            return self.raw.IndicatorLED
+        else:
+            return ""
+
+    @property
+    def num_servers(self):
+        """Retrieve the number of physical servers within this chassis."""
+        if "ComputerSystems" in self.raw.Links:
+            return len(self.raw.Links.ComputerSystems)
```

Comment:
Considering using getattr to flatten this a bit:

```
return len(getattr(self.raw.Links, 'ComputerSystems', []))
```

Reply:
👍 
I did the same in `def led_state`.